src/components/TextEditor/components/TextEditor.js

```diff
@@ -169,28 +151,62 @@ class TextEditor extends React.Component {
           return res.json();
         })
         .then((json) => {
-          const { uid, ...programData } = json;
-          this.setState({ forking: false, forked: true });
-          this.props.addProgram(uid, programData || {});
+          const { uid2, ...programData } = json;
+          setForking(false);
+          setForked(true);
+          addProgram(uid2, programData || {});
         })
-        .catch((err) => {
-          this.setState({
-            error: 'Failed to create sketch, please try again later',
-          });
-          console.log(err);
+        .catch(() => {
+          // console.log(err);
+          throw new Error('Failed to create sketch, please try again later');
         });
     } catch (err) {
-      console.log(err);
+      // console.log(err);
     }
   };
-
-  redirectSketch = () => {
-    this.closeForkModal();
-    this.setState({ redirectToSketch: true });
+  const forkModalReturn = () => {
```

Comment:
`forkModalReturn` is kind of an unclear name. Maybe something like `renderForkModalActions` would be better?